packages/flutter/lib/src/material/material.dart

```diff
@@ -548,76 +543,77 @@ class _MaterialState extends State<Material> with TickerProviderStateMixin {
   }
 }
 
-class _RenderInkFeatures extends RenderProxyBox implements MaterialInkController {
-  _RenderInkFeatures({
-    RenderBox? child,
-    required this.vsync,
-    required this.absorbHitTest,
-    this.color,
-  }) : super(child);
+/// A [RenderBox] that paints [Splash]es.
+///
+/// Splash effects can be painted on an ancestor [Material] widget by obtaining
+/// an instance of this controller via [Material.of] and calling [addSplash].
+class SplashController extends RenderProxyBox {
```

Comment:
Any benefit to moving this to its own file now that it's public?

Reply:
Unfortunately, I believe we'll have to wait until we migrate to **SplashBox**, since **_InkFeatures** needs access to **SplashController**'s private methods.